src/services/widgets/widgets.svc.coffee

```diff
@@ -74,22 +74,52 @@ angular
         setting.initialize()
       return true
 
-    @updateWidgetSettings = (widget, needContentReload=true) ->
+    @updateWidgetSettings = (widget, needContentReload=true, ignoreReach=false) ->
       widget.isEditMode = false
-
       if _.isEmpty(widget.settings)
         $log.warn("ImpacWidgetsSvc: Tried to update widget: #{widget.id} with no settings", widget)
         return false
 
+      # Check if widget settings should be written to all same widgets and use as a default
+      changedGlobalSetting = _.find widget.settings, (setting)-> setting.reach == 'dashboard'
+
+      # Update all same widgets and dashboard settings if needed
+      if (changedGlobalSetting && !ignoreReach)
+        return _self.updateAllSameWidgets(widget, ImpacDashboardsSvc.getCurrentDashboard(), changedGlobalSetting)
+
       widget.isLoading = true if needContentReload
       meta = _.reduce(_.map(widget.settings, (set) -> set.toMetadata() ), (result, setMeta) -> angular.merge(result, setMeta))
-      
-      return _self.update(widget, { metadata: meta }).then(
+
+      _self.update(widget, { metadata: meta }).then(
         (updatedWidget) ->
           if needContentReload
             _self.show(updatedWidget).finally( -> updatedWidget.isLoading = false )
       )
 
+    # TODO: move logic in ImpacDashboardsSvc
+    # Sets setting for all widgets with same name
+    @updateAllSameWidgets = (widget, dashboard, settings) ->
+      # Find name of parameter
+      paramName = _.keys(settings.toMetadata())[0];
+
+      # Find widgets with same name
+      sameWidgets = _.filter dashboard.widgets, (wgt)-> wgt.name == widget.name
```

Comment:
As explained before: you cannot test the widget's name to identify the widgets that have to be changed, because the widget's name can be changed by the user.

Try something like that:

```coffeescript
@updateAllSameWidgets = (dashboard, setting) ->
      # Find name of parameter
      paramName = _.keys(setting.toMetadata())[0];

      # Write new settings metadata to current dashboard
      ImpacDashboardsSvc.update(dashboard.id, setting.toMetadata()).then (updatedDashboard)->

        # Update new setting on current dashboard
        # TODO: can be moved to DashboardsSvc.update?
        dashboard[paramName] = updatedDashboard[paramName]

        # Set new setting to the same widgets and update them
        _.each dashboard.widgets, (wgt) ->
          # Retrieve the name of parameters attached to the widget
          # TODO: export to a helper function in WidgetsSvc
          wgtSettingsKeys = _.uniq( _.map( wgt.settings, (st) ->
            _.keys(st.toMetadata())[0]
          ))

          # The widget's metadata are updated only if the correct setting is attached to the widget
          if _.includes(wgtSettingsKeys, paramName)
            wgt.metadata[paramName] = _.cloneDeep setting.toMetadata()[paramName]
            wgt.isLoading = true
            _self.update(wgt, { metadata: wgt.metadata }).then(
              (updatedWidget) ->
                _self.show(updatedWidget).finally( -> updatedWidget.isLoading = false )
            )
```